### PR TITLE
Build mono runtime under none desktop Windows API family, adjustments and cleanup.

### PR DESCRIPTION
**Pull request by lateralusX**
Continuation work on previous none desktop Windows API family PR's adjusting all changes to similar patterns used over previous PR's. This PR also move the logic to silence a MS VS linker warning on empty source files into a macro. It also includes the use of the macro in a couple of more source files causing linker warnings when build on none desktop API families excluding JIT support.

Since other Windows API family work is still in progress it also disables build capabilities for none desktop API families.

### CODE REVIEW

File: eglib/src/gmisc-win32-uwp.c [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+		buf[0] = L'\0';
+
+	return u16to8 (buf);
+}
+
+#else /* G_HAVE_API_SUPPORT(HAVE_UWP_WINAPI_SUPPORT) */
+
+#ifdef _MSC_VER
+// Quiet Visual Studio linker warning, LNK4221, in cases when this source file intentional ends up empty.
+void __mono_win32_gmisc_win32_uwp_quiet_lnk4221(void) {}
```
Comment (ntherning) [inline eglib/src/gmisc-win32-uwp.c]: Use MONO_EMPTY_SOURCE_FILE macro here?

File: eglib/src/gmodule-win32-uwp.c [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+	ret = u16to8 (buf);
+	return ret;
+}
+
+#else /* G_HAVE_API_SUPPORT(HAVE_UWP_WINAPI_SUPPORT) */
+
+#ifdef _MSC_VER
+// Quiet Visual Studio linker warning, LNK4221, in cases when this source file intentional ends up empty.
+void __mono_win32_gmodule_win32_uwp_quiet_lnk4221(void) {}
```
Comment (ntherning) [inline eglib/src/gmodule-win32-uwp.c]: Use MONO_EMPTY_SOURCE_FILE macro here?

File: eglib/src/gunicode-win32-uwp.c [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+		*charset = my_charset;
+
+	return is_utf8;
+}
+
+#else /* G_HAVE_API_SUPPORT(HAVE_UWP_WINAPI_SUPPORT) */
+
+#ifdef _MSC_VER
+// Quiet Visual Studio linker warning, LNK4221, in cases when this source file intentional ends up empty.
+void __mono_win32_gunicode_win32_uwp_quiet_lnk4221(void) {}
```
Comment (ntherning) [inline eglib/src/gunicode-win32-uwp.c]: Use MONO_EMPTY_SOURCE_FILE macro here?

File: eglib/src/gunicode-win32.c [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+		*charset = my_charset;
+
+	return is_utf8;
+}
+
+#else /* G_HAVE_API_SUPPORT(HAVE_CLASSIC_WINAPI_SUPPORT) */
+
+#ifdef _MSC_VER
+// Quiet Visual Studio linker warning, LNK4221, in cases when this source file intentional ends up empty.
+void __mono_win32_mono_gunicode_win32_quiet_lnk4221(void) {}
```
Comment (ntherning) [inline eglib/src/gunicode-win32.c]: Use MONO_EMPTY_SOURCE_FILE macro here?

File: eglib/src/gmodule-win32-uwp.c (hunk 2 above)
Comment (lateralusX) [inline eglib/src/gmodule-win32-uwp.c]: See above comment.

**Comment by lateralusX** [comment]
Updated to latest master + changes needed to build on none Windows desktop with changes done to mono master.

**Pull request merged**